// @nodegui/nodegui/dist/lib/QtGui/QFontMetrics.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.QFontMetrics = void 0;
const addon_1 = __importDefault(require("../utils/addon"));
const Component_1 = require("../core/Component");
const helpers_1 = require("../utils/helpers");
const QSize_1 = require("../QtCore/QSize");
class QFontMetrics extends Component_1.Component {
    constructor(arg) {
        super();
        if (helpers_1.checkIfNativeElement(arg)) {
            this.native = arg;
        }
        else if (arg instanceof QFontMetrics) {
            this.native = arg.native;
        }
        else {
            this.native = new addon_1.default.QFontMetrics(arg.native);
        }
    }
    /** Returns the ascent of the font */
    ascent() {
        return this.native.ascent();
    }
    /** Returns the average width of glyphs in the font */
    averageCharWidth() {
        return this.native.averageCharWidth();
    }
    /** Returns the cap height of the font */
    capHeight() {
        return this.native.capHeight();
    }
    /** Returns the descent of the font */
    descent() {
        return this.native.descent();
    }
    /** Returns the font DPI */
    fontDpi() {
        return this.native.fontDpi();
    }
    /** Returns the height of the font */
    height() {
        return this.native.height();
    }
    /** Returns the horizontal advance in pixels of the first len characters of text. If len is negative (the default), the entire string is used */
    horizontalAdvance(text, len = -1) {
        return this.native.horizontalAdvance(text, len);
    }
    /** Returns true if character ch is a valid character in the font; otherwise returns false */
    inFont(text) {
        return this.native.inFont(text);
    }
    /** Returns the leading of the font */
    leading() {
        return this.native.leading();
    }
    /** Returns the left bearing of character ch in the font */
    leftBearing(text) {
        return this.native.leftBearing(text);
    }
    /** Returns the distance from one base line to the next */
    lineSpacing() {
        return this.native.lineSpacing();
    }
    /** Returns the width of the underline and strikeout lines, adjusted for the point size of the font */
    lineWidth() {
        return this.native.lineWidth();
    }
    /** Returns the distance from the base line to where an overline should be drawn */
    overlinePos() {
        return this.native.overlinePos();
    }
    /** Returns the right bearing of character ch in the font */
    rightBearing(text) {
        return this.native.rightBearing(text);
    }
    /**
     * Returns the size in pixels of text
     *
     * See QtEnums::TextFlag for flags
     */
    size(flags, text, tabStops = 0) {
        const native = this.native.size(flags, text, tabStops);
        return new QSize_1.QSize(native);
    }
    /** Returns the distance from the base line to where the strikeout line should be drawn */
    strikeOutPos() {
        return this.native.strikeOutPos();
    }
    /** Swaps metrics other with this metrics. This operation is very fast and never fails */
    swap(other) {
        return this.native.swap(other.native);
    }
    /** Returns the distance from the base line to where an underscore should be drawn */
    underlinePos() {
        return this.native.underlinePos();
    }
}
exports.QFontMetrics = QFontMetrics;
