// @nodegui/nodegui/dist/lib/QtWidgets/QDesktopWidget.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.QDesktopWidget = void 0;
const QRect_1 = require("../QtCore/QRect");
const QWidget_1 = require("./QWidget");
const addon_1 = __importDefault(require("../utils/addon"));
class QDesktopWidget extends QWidget_1.NodeWidget {
    constructor(parent) {
        let native;
        if (parent) {
            native = new addon_1.default.QDesktopWidget(parent.native);
        }
        else {
            native = new addon_1.default.QDesktopWidget();
        }
        super(native);
        this.native = native;
        this.nodeParent = parent;
    }
    availableGeometry(screen = -1) {
        return new QRect_1.QRect(this.native.availableGeometry(screen));
    }
    screenGeometry(screen = -1) {
        return new QRect_1.QRect(this.native.screenGeometry(screen));
    }
    screenNumber() {
        return this.native.screenNumber();
    }
}
exports.QDesktopWidget = QDesktopWidget;
